Rewrite bundle

miguel250 opened the pull request:
##### Breaks old version
###### Features
- Remove the need for https://github.com/Znarkus/postmark-php
- Use buzz to handle http requests.
- Easier to update postmark api
- Add composer
- Fix #6
###### Missing:
- Documenting(comments)
- Unit tests
- Batching messages
- Attachments 
###### Example:

``` php
<?php
$message = $this->get('postmark');
$message->addTo('<test@gmail.com>', 'Test');
$message->setSubject('subject');
$message->setTextMessage('email body');
$message->send();
```

miguel250 committed:
initiating rewrite

miguel250 committed:
Fix composer name

ruudk commented on Resources/config/services.xml:
```diff
@@ -0,0 +1,23 @@
+    </parameters>
+
+    <services>
+        <service id="postmark" class="%mz_postmark.postmark.main.class%">
```
Would it be possible to set the service as shared=false? That way, you get a new instance of Postmark\Main when you call 'postmark' service from the DIC.

phidah commented:
Awesome :)

ruudk commented:
When will this be ready and merged? :)

ruudk commented:
@miguel250 Wouldn't it be better to just make different services in the DIC? 

```
<?php
$message = $this->get('postmark.message');
$batch = $this->get('postmark.batch');
$bounces = $this->get('postmark.bounces');
```

I don't like the Uppercased methods you've created.

ruudk commented:
If you create separate services you can inject the base 'postmark' service to all of them. That service contains the API key etc. 'postmark.message' would then be a single message instance as it returns a new instance on every call.

miguel250 commented:
@ruudk I won't have time to work on this until next week but pull requests are welcome against the rewrite branch.

ruudk commented:
I also think the "PostMark" directory should be called "Postmark" as that's the official name of the company.

miguel250 committed:
Remane postmark folder

miguel250 committed:
Lock buzz to version 0.7

miguel250 committed:
Refractor service name and class names

Rename postmark service to postmark.message and create HTTPClient class

miguel250 committed:
Add comments

miguel250 committed:
Remove fromName.

Use from_email: "name <email>" instead

miguel250 committed:
Add author document tag and fix cs

miguel250 committed:
Use dev-master as buzz version until version with fix to headers is created

https://github.com/kriswallsmith/Buzz/commit/8d6d5da0ec38d656cf42297a99e89f271763789c

miguel250 committed:
Add unit tests

miguel250 committed:
Add support to xml configuration

miguel250 committed:
Fix cs

miguel250 committed:
Add read me file
